gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionFactory.java

```diff
@@ -432,16 +493,49 @@ protected void setElementConverter(JsonElementConverter elementConverter) {
     public JsonElementConverter getElementConverter() {
       return this.elementConverter;
     }
+
+    protected void processNestedItems(JsonSchema schema, WorkUnitState state)
+        throws UnsupportedDateTypeException {
+      JsonSchema nestedItem = null;
+      if (schema.isType(ARRAY)) {
+        nestedItem = schema.getItemsWithinDataType();
+      }
+      if (schema.isType(MAP)) {
+        nestedItem = schema.getValuesWithinDataType();
+      }
+      this.setElementConverter(getConvertor(nestedItem, null, state));
+    }
+
+    @Override
+    Object convertField(JsonElement value) {
```

Comment:
Are these 3 methods (convertField, getTargetType, and schema) meant to be in ComplexConverter? They seem like they are specific to ArrayConverter and already exist inside it.